internal/handlers/pg/cmd_query.go

```diff
@@ -16,47 +16,71 @@ package pg
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"time"
+	"strings"
+
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
 
 	"github.com/FerretDB/FerretDB/internal/handlers/common"
 	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
 	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // CmdQuery implements HandlerInterface.
 func (h *Handler) CmdQuery(ctx context.Context, query *wire.OpQuery) (*wire.OpReply, error) {
-	if query.FullCollectionName == "admin.$cmd" {
-		switch cmd := query.Query.Command(); cmd {
-		case "ismaster", "isMaster": // both are valid
-			reply := &wire.OpReply{
-				NumberReturned: 1,
-				Documents: []*types.Document{must.NotFail(types.NewDocument(
-					"ismaster", true, // only lowercase
-					// topologyVersion
-					"maxBsonObjectSize", int32(types.MaxDocumentLen),
-					"maxMessageSizeBytes", int32(wire.MaxMsgLen),
-					"maxWriteBatchSize", int32(100000),
-					"localTime", time.Now(),
-					// logicalSessionTimeoutMinutes
-					"connectionId", int32(42),
-					"minWireVersion", common.MinWireVersion,
-					"maxWireVersion", common.MaxWireVersion,
-					"readOnly", false,
-					"ok", float64(1),
-				))},
+	cmd := query.Query.Command()
+	collection := query.FullCollectionName
+
+	// both are valid
+	if (cmd == "ismaster" || cmd == "isMaster") && collection == "admin.$cmd" {
+		return common.IsMaster(ctx, query)
+	}
+
+	// defaults to the database name if supplied on the connection string or $external
+	if cmd == "saslStart" &&
```

Comment:
Codecov thinks below this line is not covered by test, could you add integration test please?

Reply:
But how to do that?

Reply:
I don't know how to do this, let me look around :see_no_evil:

Reply:
Yeah you are right, Ops query is only used for old version so our tests won't hit this